refactor!: temporary deprecate positional arguments of `pl$Series()` (#966)
pola-rs/r-polars: NEWS.md

### Other breaking changes

- Several functions have been rewritten to match the behavior of Python Polars.
  - In `pl$Series()` arguments are changed.
    - The argument converted to the values of the Series is renamed from `x` to `values` (#933).

>     - Using positional arguments in `pl$Series()` is warned, since the

Reviewer: ```suggestion
    - Using positional arguments in `pl$Series()` throws a warning, since the
```

    - Using positional arguments in `pl$Series()` throws a warning, since the
      argument positions will be changed in the future (#966).

      ```r
      # polars 0.15.1 or earlier
      # The first argument is `x`, the second argument is `name`.
      pl$Series(1:3, "foo")

      # The code above will warn in 0.16.0
      # Use named arguments to silence the warning.
      pl$Series(values = 1:3, name = "foo")
      pl$Series(name = "foo", values = 1:3)

      # polars 0.17.0 or later (future version)
      # The first argument is `name`, the second argument is `values`.
      pl$Series("foo", 1:3)

>       ```

Reviewer: ```suggestion
      ```
      This warning can also be silenced by replacing `pl$Series(<values>, <name>)` 
      by `as_polars_series(<values>, <name>)`.
```

Reviewer: We should advertise `as_polars_series()` as a way to silence this warning

>       ```r
>       # polars 0.15.1 or earlier
>       # The first argument is `x`, the second argument is `name`.
>       pl$Series(1:3, "foo")
>       ```
>       ```r
>       # polars 0.17.0 or later (future version)
>       # The first argument is `values`, the second argument is `name`.
>       pl$Series("foo", 1:3)
>       ```
>       To silence the warning, please use named arguments.
>       ```r
>       # polars 0.16.0 or later
>       pl$Series(values = 1:3, name = "foo")
>       pl$Series(name = "foo", values = 1:3)
>       ```

Reviewer: ```suggestion
      ```r
      # polars 0.15.1 or earlier
      # The first argument is `x`, the second argument is `name`.
      pl$Series(1:3, "foo")
      # The code above will warn in 0.16.0
      # Use named arguments to silence the warning.
     
      # polars 0.16.0 or later
      pl$Series(values = 1:3, name = "foo")
      pl$Series(name = "foo", values = 1:3)
      
      # polars 0.17.0 or later (future version)
      # The first argument is `name`, the second argument is `values`.
      pl$Series("foo", 1:3)
```

Reviewer: Reordered the chunks to have 0.15.0 then 0.16.0 then 0.17.0. Put everything in a single chunk
